### ISSUE TITLE
Training my own dataset - Error

### ISSUE THREAD

**Issue by fjfo** [D-X-Y/caffe-faster-rcnn#9]
### Issue summary
- I create my own train_val.prototxt with my own solver file.
- And then prepare my own training text file with formated (as you said in Document) like:
```
# number
Filename
count of box
label x1 y1 x2 y2 dif
...

```- And then run file: train_frcnn.sh you provided in Example folder without pre-trained weight of ZF or VOC. Because I use my own prototxt that does not fit ZF or VOC model.

- I can see training log file, thats good and finish after 70000 loop.
No any error at all in training step.
But
When I get final model after training to test.
This trained-model can not predict any box or label, 100% fail, even with training files.
Why that? Do I miss any step in training?

### Your system configuration
Operating system: Ubuntu 16.04
CUDA version (if applicable): 8.0
CUDNN version (if applicable): 5.1

**Comment by D-X-Y** [comment]
Do you use examples/FRCNN/convert_frcnn_model.sh to convert you trained model to that can be used for test.
The box regression has a normalization procedure, which is automatically done in my code.
So before testing, we should do a little change on the bbox prod layer, appending this normalization information.

**Comment by fjfo** [comment]
oh, I can see "convert_frcnn_model" in your "examples/FRCNN/train_frcnn.sh" like this:
`#!/usr/bin/env sh
# This script test four voc images using faster rcnn end-to-end trained model (ZF-Model)
if [ ! -n "$1" ] ;then
    echo "$1 is empty, default is 0"
    gpu=0
else
    echo "use $1-th gpu"
    gpu=$1
fi

CAFFE=build/tools/caffe 

time GLOG_log_dir=examples/FRCNN/log $CAFFE train   \
    --gpu $gpu \
    --solver models/FRCNN/zf/solver.prototxt \
    --weights models/FRCNN/ZF.v2.caffemodel 

time python examples/FRCNN/convert_model.py \
    --model models/FRCNN/zf/test.prototxt \
    --weights models/FRCNN/snapshot/zf_frcnn_end_to_end_iter_70000.caffemodel \
    --config examples/FRCNN/config/voc_config.json \
    --net_out models/FRCNN/zf_faster_rcnn_final.caffemodel
`
First, you train with solver file and then you convert to final model.
After converting, I use this final model to test.

**Comment by D-X-Y** [comment]
So, you train the model and convert it for test phase, but the visalized results are wrong?
I will check this soon.

**Comment by fjfo** [comment]
yes, 100% wrong.

If I use ZF or VOC pre-trained weights and model, it will be ok.
Output with good bbox and cls.

But training from scratch, it will be wrong all the time.

**Comment by D-X-Y** [comment]
If using the pre-train model is ok, then it means the code is ok.
I hae tried this code on face detection, the results are ok.
If training from scratch is wrong, do you check your learning rate or some other parameters.

**Comment by fjfo** [comment]
I trained in 2 cases:
Case 1: Train with pre-train model and weight of ZF
Data input: my own training data files
Result: BBox and Cls are good.

Case 2: Same config, same Lr, same parameters, same models, same data input with ZF with small modified in layer "bbox_pred" from "num_output: 84" to "num_output: 8"
and layer "cls_score" from "num_output: 21"to "num_output: 2"

 I just want to use 2 classes and 2*4=8 coordinate in predicted-box.
Training time: bbox_accuracy=0.99, bbox_loss=3e-05,cls_loss=0.1
After 70000 loops, this final model can not predict any box, event with training image files?
T_T

Can you suggest anything to fix?

**Comment by D-X-Y** [comment]
There is a configure file, you should also change it.

**Comment by fjfo** [comment]
yes, ofcourse
I changed "n_classes: 2"

**Comment by xwg031459** [comment]
please, where is the dir of "examples/FRCNN" ? why not find.

**Comment by D-X-Y** [comment]
@xwg031459  Please check into the 'dev' branch.

**Comment by D-X-Y** [comment]
@fjfo  O.O  Maybe you could provide more information about the dataset and the prototxt.?
Is it possible to overfit the training set?

**Comment by fjfo** [comment]
hi @D-X-Y 
I give you my own dataset and prototxt, config file in link
[http://www.mediafire.com/file/i9u2v9mjn3xphl7/model.tar.gz](url)

You can use: train_frcnn.sh script to train it from scratch to check it out.
I can sent you my final model with pre-trained weight ZF to compare the result.

Thank,

**Comment by jjn037** [comment]
@fjfo have u got a good results from a finetue model? I  have tried to use this code but got a bad result

**Comment by fjfo** [comment]
@jjn037  yes, this code had good result with pre-trained model.

**Comment by dainylucy** [comment]
@fjfo @D-X-Y I just want to train my own dataset with pretrained VGG16 model ,for 2 classes.
and  I modified the train_val.prototxt  with the layer  "bbox_pred" from "num_output: 84" to "num_output: 8" and layer "cls_score" from "num_output: 21"to "num_output: 2",and also changed "n_classes" to "2".
when I trained ,the error message is  cannot copy param 0 weights from layer :''cls_score'';shape missmatch.......................................
why?how shold I solve the problem?
